### PR TITLE
Fix issue with ARCore and improve camera capturer

### PR DESCRIPTION
**Pull request by callum-code**
I'd ran into this bug shortly after I started working on this and only just now cracked it. This unblocked me to fix a few things with the AR Core scene image capturer.

### CODE REVIEW

File: iOS/MMIWG2S/Utilities/Extensions.swift [hunk 1]
```diff
@@ -99,3 +121,7 @@ extension UIView {
     }
     
 }
+
+public func * (size: CGSize, scalar: CGFloat) -> CGSize {
+    return CGSize(width: size.width * scalar, height: size.height * scalar)
+}
```
Comment (Mehequanna) [inline iOS/MMIWG2S/Utilities/Extensions.swift:127]: Well that's cool and also odd you had to create your own multiplication function.

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift [hunk 2]
```diff
@@ -206,25 +221,42 @@ public final class FacesViewController: UIViewController {
...
+        isCapturing = true
         let deviceOrientation = UIDevice.current.orientation
         guard let pixelBuffer = faceSession?.currentFrame?.capturedImage,
-              let image = UIImage(pixelBuffer: pixelBuffer, orientation: .from(deviceOrientation: deviceOrientation))
-              else { return nil }
+              let capturedImage = UIImage(pixelBuffer: pixelBuffer, orientation: .from(deviceOrientation: deviceOrientation))
+              else {
+            isCapturing = false
+            return nil
+        }
```
Comment (Mehequanna) [inline iOS/MMIWG2S/ViewControllers/FacesViewController.swift:239]: ❓ This is really weird spacing. Is it normal?

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift [hunk 3]
```diff
@@ -206,25 +221,42 @@ public final class FacesViewController: UIViewController {
...
+        
         captureSession?.stopRunning()
-        return image
+        var sceneImage = sceneView.snapshot()
+        
+        if deviceOrientation.isLandscape {
+            sceneImage = sceneImage.rotate(radians: deviceOrientation == .landscapeLeft ? .pi / 2 : -.pi / 2) ?? sceneImage
+        }
```
Comment (Mehequanna) [inline iOS/MMIWG2S/ViewControllers/FacesViewController.swift:246]: ❓ The app is always in portrait, correct? Is this needed? Keep it for sure, just curious.

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift [hunk 4]
```diff
@@ -206,25 +221,42 @@ public final class FacesViewController: UIViewController {
...
+        UIGraphicsBeginImageContextWithOptions(capturedImage.size, true, 0.0)
+        capturedImage.draw(in: CGRect(origin: .zero, size: capturedImage.size))
+        sceneImage.draw(in: CGRect(origin: .zero, size: capturedImage.size))
```
Comment (Mehequanna) [inline iOS/MMIWG2S/ViewControllers/FacesViewController.swift:249]: How do these gets used? I see they are drawn, just not sure how they are used :)

**Review by Mehequanna — APPROVED** [review]
Just a few questions. Thanks for doing this!

File: iOS/MMIWG2S/Utilities/Extensions.swift (hunk 1 above)
Comment (callum-code) [reply iOS/MMIWG2S/Utilities/Extensions.swift:127]: yeah swift only provides the basics by default

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift (hunk 2 above)
Comment (callum-code) [reply iOS/MMIWG2S/ViewControllers/FacesViewController.swift:239]: the `else` could be deindented. Some ppl do it inline with the end of the `guard`, some on the outer layer.

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift (hunk 3 above)
Comment (callum-code) [reply iOS/MMIWG2S/ViewControllers/FacesViewController.swift:246]: Currently I don't believe we have it locked to portrait for iPad. Either way, just being safe for the future.

File: iOS/MMIWG2S/ViewControllers/FacesViewController.swift (hunk 4 above)
Comment (callum-code) [reply iOS/MMIWG2S/ViewControllers/FacesViewController.swift:249]: its drawn to a static instance. `UIGraphicsGetImageFromCurrentImageContext` reads from the current context we were just writing to

**Pull request merged**